File: schemas/digital-asset-management/drizzle/collection_assets.ts
// collection_assets: Join table linking assets to collections with ordering.
// See README.md for full design rationale.

import { pgTable, uuid, integer, timestamp, index, unique } from "drizzle-orm/pg-core";
import { users } from "../../auth-rbac/drizzle/users";
import { collections } from "./collections";
import { assets } from "./assets";

export const collectionAssets = pgTable(
  "collection_assets",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    collectionId: uuid("collection_id").notNull().references(() => collections.id, { onDelete: "cascade" }),
    assetId: uuid("asset_id").notNull().references(() => assets.id, { onDelete: "cascade" }),
    position: integer("position").notNull().default(0),
    addedBy: uuid("added_by").notNull().references(() => users.id, { onDelete: "restrict" }),
    addedAt: timestamp("added_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    unique("uq_collection_assets_collection_id_asset_id").on(table.collectionId, table.assetId),
    index("idx_collection_assets_asset_id").on(table.assetId),
  ]
);

File: schemas/digital-asset-management/spacetimedb/collection_assets.rs
// collection_assets: Join table linking assets to collections with ordering.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

// Composite unique(collection_id, asset_id) — enforce in reducer logic

#[spacetimedb::table(name = collection_assets, public)]
pub struct CollectionAsset {
    #[primary_key]
    pub id: String, // UUID

    #[index(btree)]
    pub collection_id: String, // FK → collections.id (cascade delete)

    #[index(btree)]
    pub asset_id: String, // FK → assets.id (cascade delete)

    pub position: i32, // default 0

    pub added_by: String, // FK → users.id (restrict delete)

    pub added_at: Timestamp,
}
